gpiozero/devices.py

```diff
@@ -229,13 +189,100 @@ def __exit__(self, exc_type, exc_value, exc_tb):
 class Device(ValuesMixin, GPIOBase):
     """
     Represents a single device of any type; GPIO-based, SPI-based, I2C-based,
-    etc. This is the base class of the device hierarchy. It defines the
-    basic services applicable to all devices (specifically the :attr:`is_active`
+    etc. This is the base class of the device hierarchy. It defines the basic
+    services applicable to all devices (specifically the :attr:`is_active`
     property, the :attr:`value` property, and the :meth:`close` method).
     """
+    _pin_factory = None # instance of a Factory sub-class
+    _reservations = {} # maps pin addresses to lists of devices
```

Comment:
I'm obviously not following what the code is doing, but why is `_reservations` stored on the Device and not the PinFactory?